Controller/Webhook/Update.php

```diff
@@ -0,0 +1,197 @@
+            $stock->setQty($product["inventory"]["available"]);
+        }
+    }
+
+    private function handleOrderShipped($data)
+    {
+        /** @var bool $result */
+        $result = false;
+
+        /** @var \Magento\Sales\Model\Order $order */
+        $order = $this->orderRepository->get($data["merchant_id"]);
+
+        //A shipment might have been created if autoship was set to true in module settings
+        if (!$order->hasShipments() && $order->canShip()) {
+            $qty = array();
+            foreach ($order->getAllItems() as $item) {
+                $qty[$item->getId()] = $item->getQtyOrdered();
+            }
+
+            /** @var \Magento\Sales\Model\Convert\Order $convertOrder */
+            $convertOrder = $this->convertOrderFactory->create();
+
+            /** @var \Magento\Sales\Model\Order\Shipment $shipment */
+            $shipment = $convertOrder->toShipment($order);
+
+            $shipment->register();
+            $shipment->addComment(__("Tracking URL: " . str_replace('\\', '', $data['tracking_url'])));
+            $track = $this->trackFactory->create();
+            $track->addData(array(
+                'carrier_code' => 'custom',
+                'title'        => 'Logitrail',
+                'number'       => $data['tracking_code']
+            ));
+            $shipment->addTrack($track);
+            $shipment->getOrder()->setState(\Magento\Sales\Model\Order::STATE_COMPLETE);
+
+            $transactionSave = $this->transactionFactory->create();
+            $transactionSave->addObject($shipment)
+                ->addObject($order)
+                ->save();
+            $this->shipmentNotifier->notify($shipment);
+
+            $result = true;
+        }
+        return $result;
+    }
+
+    private function authenticate(\Zend\Http\Headers $headers)
+    {
+        $headerArray = $headers->toArray();
+
+        if (!isset($headerArray["Authorization"])) {
+            //No auth field in headers so can't authenticate the request. stop processing
+            return false;
+        }
+
+        // The auth field from the header
+        $authString = $headerArray["Authorization"];
+
+        // Split the string into the auth type and auth string
+        $authArray = explode(" ", $authString);
+
+        // Check the type is "Basic"
+        if ($authArray[0] != "Basic") {
+            // Not something we are expecting so stop processing.
+            return false;
+        }
+
+        // Decode the auth string and
+        $auth = explode(':', base64_decode($authArray[1]));
+
+        // Get the local values for authentication
+        $webhooksname = $this->scopeConfig->getValue('carriers/logitrail/webhooksname', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
+        $webhookskey = $this->scopeConfig->getValue('carriers/logitrail/webhookskey', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
+
+        // If the first value doesn't match return false
```

Comment:
Comment should clarify what is happening in the code. "First value" doesn't clarify a lot. Might be useful to actually tell what is the first value that is compared here, and second in the next if.